app/routes.py

```diff
@@ -390,3 +399,47 @@ def forums():
 @app.route('/profile')
 def profile():
     return render_template('profile.html', title='CONP | My Profile', user=current_user)
+
+@app.route('/pipeline-search', methods=['GET'])
+def pipeline_search():
+    if request.method == 'GET':
+
+        #initialize variables
+        search_query = request.args.get("search")
+        cache_dir = os.path.join(os.path.expanduser('~'), ".cache", "boutiques")
+        all_desc_path = os.path.join(cache_dir, "all_descriptors.json")
+        all_detailed_desc_path = os.path.join(cache_dir, "detailed_all_descriptors.json")
+
+        #fetch data from cache
+        with open(all_desc_path, "r") as f:
+            all_descriptors = json.load(f)
+
+        with open(all_detailed_desc_path, "r") as f:
+            detailed_all_descriptors = json.load(f)
```

Comment:
Isn't there a way to keep that in memory? Now you're going to reload all these files with every query.

Reply:
From my knowledge, if you want to use threads to update data in the background in memory, then there is no other choice than to declare those data variables global which is bad practice because I ran a few tests and I couldn't pass the data variables to the threads to be updated as the threads create local copies of those variables and update those instead and there is no way to return those updated data variables as we are using threads except via caching in files to be read.

I think we have few options to go with:
- Either keep the code like this, meaning we will have to read from the cache everytime there is a query, but at least we can use threads to update data in the background and there is no global variable
- We can stop using threads meaning that the query that triggers the data update will take a little bit longer to be responded to since it will have to wait for the updateData() method until the local data is fully updated.
- Setup some kind of database we can query from and that is self updated/updated by threads, but this options is pretty similar from the first option as they both read/query from a third party unless I am wrong.

Let me know if there is any better options

Reply:
Option 1 seems the best for now. Option 3 is definitely the way to go when we want to improve performance. Thanks for looking into this!